server/threads/sql/index.ts

```diff
@@ -30,14 +27,13 @@ const getRoleByExternalIdAndBoardId = `
     LEFT JOIN board_user_roles bur
       ON roles.id = bur.role_id
     LEFT JOIN realm_user_roles rur
-      ON roles.id = rur.role_id
+      ON roles.id = rur.role_id AND rur.realm_id = (SELECT parent_realm_id FROM boards WHERE boards.string_id = $/board_external_id/)
     INNER JOIN users
-      ON users.id = bur.user_id  OR users.id = rur.user_id
+      ON (users.id = bur.user_id OR bur.user_id IS NULL) AND (users.id = rur.user_id OR rur.user_id IS NULL)
```

Comment:
I don't understand this change

Reply:
The logic was wrong earlier, but I rewrote the query again so it's all much more understandable.